frontend/public/components/volumes-table.tsx

```diff
@@ -5,44 +5,58 @@ import * as classNames from 'classnames';
 
 import {
   ContainerSpec,
+  getVolumeType,
+  getVolumeLocation,
+  K8sResourceKind,
   PodKind,
   PodTemplate,
   Volume,
   VolumeMount,
 } from '../module/k8s';
-import {
-  getVolumeType,
-  getVolumeLocation,
-} from '../module/k8s/pods';
-import {
-  VolumeIcon,
-  ResourceIcon,
-  EmptyBox,
-  SectionHeading,
-} from './utils';
+import { VolumeIcon, ResourceIcon, EmptyBox, SectionHeading, VolumeKebab, Kebab, asAccessReview, VolumeKebabFactory } from './utils';
+import { Table, TableData, TableRow } from './factory';
+import { sortable } from '@patternfly/react-table';
+import { removeVolumeModal } from './modals';
 
-const getVolumeMountsByPermissions = (pod: PodTemplate): VolumeData[] => {
+const VolumeKebabMenu: VolumeKebabFactory = {
+  RemoveVolume: (kind, obj, volume) => ({
+    label: 'Remove Volume',
+    callback: () => removeVolumeModal({
+      kind,
+      resource: obj,
+      volume,
+    }),
+    accessReview: asAccessReview(kind, obj, 'patch'),
+  }),
+};
+
+const { RemoveVolume } = VolumeKebabMenu;
+const menuActions = [RemoveVolume];
+
+const getPodTemplate = (resource: K8sResourceKind): PodTemplate => {
+  return resource.kind === 'Pod'? resource: resource.spec.template;
```

Comment:
Surprised this isn't a TypeScript error

```suggestion
  return resource.kind === 'Pod' ? resource as PodKind : resource.spec.template;
```